crates/variables/src/provider/azure_key_vault.rs

```diff
@@ -2,54 +2,113 @@ use std::sync::Arc;
 
 use anyhow::{Context, Result};
 use async_trait::async_trait;
+use azure_core::auth::TokenCredential;
 use azure_core::Url;
 use azure_security_keyvault::SecretClient;
 use serde::Deserialize;
 use spin_expressions::{Key, Provider};
 use tracing::{instrument, Level};
 
-#[derive(Debug)]
-pub struct AzureKeyVaultProvider {
+/// Azure KeyVault runtime config literal options for authentication
+#[derive(Clone, Debug)]
+pub struct AzureKeyVaultRuntimeConfigOptions {
     client_id: String,
     client_secret: String,
     tenant_id: String,
-    vault_url: String,
     authority_host: AzureAuthorityHost,
 }
 
-impl AzureKeyVaultProvider {
+impl AzureKeyVaultRuntimeConfigOptions {
     pub fn new(
-        client_id: impl Into<String>,
-        client_secret: impl Into<String>,
-        tenant_id: impl Into<String>,
-        vault_url: impl Into<String>,
-        authority_host: impl Into<AzureAuthorityHost>,
+        client_id: String,
+        client_secret: String,
+        tenant_id: String,
+        authority_host: Option<AzureAuthorityHost>,
     ) -> Self {
         Self {
-            client_id: client_id.into(),
-            client_secret: client_secret.into(),
-            tenant_id: tenant_id.into(),
-            vault_url: vault_url.into(),
-            authority_host: authority_host.into(),
+            client_id,
+            client_secret,
+            tenant_id,
+            authority_host: authority_host.map_or(AzureAuthorityHost::default(), |ah| ah),
```

Comment:
If I'm reading this right, it would be more idiomatic to use `.unwrap_or_else(|| AzureAuthorityHost::default()` rather than passing things through the identity function.  But I might be missing something!